### ISSUE TITLE
How to reach optimal performance with stream?

### ISSUE THREAD

**Issue by glyh** [jecolon/ziglyph#17]
As title. In the source code I see this: https://github.com/jecolon/ziglyph/blob/531d30951219f4703c85a51521745860185be6f2/src/segmenter/Grapheme.zig#L244

Which is trying to break down the text by lines. However, if the input is so long in one line, this will probably be problematic that we can't store the whole line in the [4096]u8; Also, if there's too many newlines, this will be inefficient, in that it will call the read function multiple times. Of course we may use a bufferedReader, but I feel there's redundant work here, i.e. we are buffering after using an already buffered Reader.

Is there any workaround that I can make this better? 

Thank you!

**Comment by glyh** [comment]
TLDR: How do I do a "Grapheme View" over a Reader without consuming any input from the buffer, efficiently?

**Comment by glyh** [comment]
NVM, after I examine the internal for the Grapheme, I feel I know how should I implement this right now
